tests/test_tasks.py

```diff
@@ -2978,6 +2979,34 @@ def my_flow():
         task_states = my_flow()
         assert [state.result() for state in task_states] == [6, 7, 8]
 
+    def test_map_with_sequential_runner_is_sequential(self):
```

Comment:
As far as making this more performant of a test, you could map over `range(10)` or something then have each task sleep for `1 / (n * 100)` to do much smaller sleeps but still have the first one sleep the longest?